src/reactjs/__tests__/App.test.js

```diff
@@ -0,0 +1,46 @@
+import { render, fireEvent, screen } from '@testing-library/react';
+import renderer from 'react-test-renderer';
+import { BrowserRouter } from 'react-router-dom';
+import '@testing-library/jest-dom';
+
+import App from '../components/App';
+
+describe('it tests the App Component', () => {
+  it('renders App component correctly', () => {
+    const tree = renderer
+      .create(<BrowserRouter><App /></BrowserRouter>)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  it('renders the App Component', () => {
+    render(<BrowserRouter><App /></BrowserRouter>);
+    const appComponent = screen.getByTestId('app-1');
+    expect(appComponent).toBeInTheDocument();
+  });
+
+  it('renders the home page in the App component when the user clicks the home link', () => {
+    render(<BrowserRouter><App /></BrowserRouter>);
+    const homeLink = screen.getByText('Home');
+    fireEvent.click(homeLink);
+    const homeContainer = screen.getByTestId('home-1');
+    const firstchild = homeContainer.childNodes[0];
```

Comment:
- [ ] [OPTIONAL] To follow best practices for JavaScript let's kindly make sure we name variable with camel case and be consistent with it I could see you did a great job at naming `homeContainer` but when it came to `firstchild` wasn't camel case this is not the only example in the project but having this in mind that would help you later on to follow best practices easily I hope ☺️